price-estimator/src/orderbook.rs

```diff
@@ -1,50 +1,98 @@
 use anyhow::Result;
-use core::{models::BatchId, orderbook::StableXOrderBookReading};
+use core::{
+    models::{AccountState, BatchId, Order},
+    orderbook::StableXOrderBookReading,
+};
 use pricegraph::Pricegraph;
 use tokio::sync::RwLock;
 
+#[derive(Debug)]
+pub enum PricegraphType {
+    // pricegraph instance with the original orders from the orderbook
+    Raw,
+    // pricegraph instance with the orders to which the rounding buffer has been applied
+    #[allow(dead_code)]
+    WithRoundingBuffer,
+}
+
 /// Access and update the pricegraph orderbook.
 pub struct Orderbook {
     orderbook_reading: Box<dyn StableXOrderBookReading>,
-    pricegraph: RwLock<Pricegraph>,
+    pricegraph_raw: RwLock<Pricegraph>,
```

Comment:
ubernit: To keep the struct more focussed the two pricegraphs could be behind a common `cache` struct and the same lock (given they are only ever written to at the same time). Right now the Orderbook struct is still small so maybe not needed.